Client/core/CIMGArchive.h

```diff
@@ -0,0 +1,60 @@
+#ifndef CIMGArchive_H
```

Comment:
Please use `#pragma once` instead

Reply:
@qaisjp I think it's too late to review the code. I'm just trying to make it work for now. I'll rewrite it when I'm done. ;p